modules/axhal/src/platform/aarch64_common/pl031.rs

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) [2023] [Syswonder Community]
+ *   [Rukos] is licensed under Mulan PSL v2.
+ *   You can use this software according to the terms and conditions of the Mulan PSL v2.
+ *   You may obtain a copy of Mulan PSL v2 at:
+ *               http://license.coscl.org.cn/MulanPSL2
+ *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+ *   See the Mulan PSL v2 for more details.
+ */
+
+//! PL031 RTC.
+
+static RTC_DR: u32 = 0x000;
+static RTC_MR: u32 = 0x004;
+static RTC_LR: u32 = 0x008;
+static RTC_CR: u32 = 0x00c;
+static RTC_IMSC: u32 = 0x010;
+static RTC_RIS: u32 = 0x014;
+static RTC_MIS: u32 = 0x018;
+static RTC_ICR: u32 = 0x01c;
+
+const PHYS_RTC: usize = axconfig::PHYS_VIRT_OFFSET + 0x09010000;
+
+static PL031_RTC: Pl031rtc = Pl031rtc { address: PHYS_RTC };
+
+pub fn init() {
+    info!("Initialize pl031 rtc...");
+    PL031_RTC.init();
+    debug!("{}", rtc_read_time());
+}
+
+struct Pl031rtc {
+    address: usize,
+}
+
+impl core::fmt::Display for Pl031rtc {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "RTC DR: {}\nRTC MR: {}\nRTC LR: {}\nRTC CR: {}\nRTC_IMSC: {}",
+            unsafe { self.read(RTC_DR) } as u64,
+            unsafe { self.read(RTC_MR) } as u64,
+            unsafe { self.read(RTC_LR) } as u64,
+            unsafe { self.read(RTC_CR) } as u64,
+            unsafe { self.read(RTC_IMSC) } as u64
+        )
+    }
+}
+
+impl Pl031rtc {
+    fn init(&self) {
+        unsafe {
+            if self.read(RTC_CR) != 1 {
```

Comment:
Is this `if` block necessary? Or use `self.write(RTC_CR, 1)` directly?

Reply:
the manual said " If set to 1, the RTC is enabled. After the RTC is enabled, do not write to this bit otherwise the current 
RTC value is reset to zero." so if it is already 1 I will not set it.